src/inversion_scripts/make_gridded_posterior.py

```diff
@@ -65,9 +65,9 @@ def make_gridded_posterior(posterior_SF_path, state_vector_path, save_path):
     lon = gridded_SF["lon"].values
     ds = xr.Dataset(
         {
-            "ScaleFactor": (["lat", "lon"], gridded_SF),
-            "S_post": (["lat", "lon"], gridded_S_post),
-            "A": (["lat", "lon"], gridded_A),
+            "ScaleFactor": (["lat", "lon"], gridded_SF.data),
+            "S_post": (["lat", "lon"], gridded_S_post.data),
+            "A": (["lat", "lon"], gridded_A.data),
```

Comment:
According to the numpy documentation `ndarray.data` is a `Python buffer object pointing to the start of the array’s data`. This is directly accessing memory and from what I have read can be error prone. I think remove the `.data` here

Reply:
That was added by @megan-he in commit 24bac7b. I am fine with removing.